build/scripts/package.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-nocheck <-- This is needed as this file is basically a JavaScript script
+//                 but with some TypeScript niceness baked in
+const fs = require('fs')
+const path = require('path')
+const glob = require('glob')
+const execSync = require('child_process').execSync
+const pkg_dir = 'dist/package'
+
+async function copyGlob(pattern) {
+  glob(pattern, (error, files) => {
+    for (var i = 0; i < files.length; i++) {
+      // Skip dirs as they are most likely listed in the .vscodeignore as well
+      if (fs.statSync(files[i]).isDirectory()) continue
```

Comment:
Why skip directories that match the glob? If a glob matches a directory in .vscodeignore don't we want them too?

Reply:
See https://github.com/apache/daffodil-vscode/pull/232#issuecomment-1189027422 for comment about that

Reply:
Would this work

```typescript
if (fs.statSync(src).isDirectory() || !fs.existsSync(dstDir)) {
    fs.mkdirSync(dstDir, { recursive: true })
    if (fs.statSync(src).isDirectory()) continue
}
```

Reply:
I'm not sure I understand the logic in that snippet. What's the reason for the existsSync check? mkdirSync with `recrusive:true` doesn't error if it already exist so shouldn't be needed?

I also generally perfer if/else instead of an if- with a continue in the block, with one block being dir logic and another block being file logic.

Reply:
Does the update made look better?